routers/api/v1/packages/package.go

```diff
@@ -213,3 +214,54 @@ func ListPackageFiles(ctx *context.APIContext) {
 
 	ctx.JSON(http.StatusOK, apiPackageFiles)
 }
+
+// LinkPackage sets a repository link for a package
+func LinkPackage(ctx *context.APIContext) {
+	// swagger:operation POST /packages/{owner}/{type}/{name}/link package linkPackage
+	// ---
+	// summary: Link a package to a repository
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the package
+	//   type: string
+	//   required: true
+	// - name: type
+	//   in: path
+	//   description: type of the package
+	//   type: string
+	//   required: true
+	// - name: name
+	//   in: path
+	//   description: name of the package
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: query
+	//   description: ID of the repository to link
+	//   type: integer
+	//   required: true
```

Comment:
Would be nice if you can't just pass the repository id (which you don't have usually) but the repo name.
Could have two parameters `repo_id=...` and `repo_name=...`. The endpoint should choose the matching repo then. Could prefer the id over the name.

Reply:
Not particularly proud of how my code parsing those two parameters turned out, but I didn't find any useful utilities.
Any ideas to improve [`221dcdc`](https://github.com/go-gitea/gitea/pull/23851/commits/221dcdc4beb7bf79bcd1ad060172800aa28d28a9)?